utils/composer.json

```diff
@@ -0,0 +1,37 @@
+{
+    "require": {
+        "php": "^8.1",
+        "codeigniter/coding-standard": "^1.7",
+        "codeigniter/phpstan-codeigniter": "^1.4",
+        "codeigniter4/codeigniter4": "4.x-dev",
```

Comment:
Why `4.x-dev`?
```console
$ composer update
...
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires codeigniter4/codeigniter4 4.x-dev (exact version match), found codeigniter4/codeigniter4[dev-reorganise-toolkit, 4.6.x-dev] but it does not match the constraint.

Script @composer update --working-dir=utils handling the post-autoload-dump event returned with error code 2
```

Reply:
4.x-dev is the branch alias of the develop branch. See composer.json for that.

This requirement is really a hack to make Psalm happy on Github Actions. Without this line, psalm would complain of DuplicateClass for **all** CodeIgniter classes. phpstan-codeigniter requires codeigniter4/framework. Without this line, psalm would not error on my local but errors on GHA. Not sure why is that.

Since this repo replaces codeigniter4/framework (see replace of composer.json), I realized I can just use this repo by using composer's `repositories` section.

Reply:
Next, I would want to specify the branch name to require. I initially thought of using the current branch but the approach would be dynamic and cannot be fixed to this composer.json. Basically, we will need `git` to get the current branch name then manually require it here and on GHA.

However, I remembered phpstan-codeigniter only uses the framework for very few of its classes, like BaseServices, `config()` and `model()` functions, etc. so I opted to just require the develop branch.

Reply:
If you want to require the current branch, example this PR branch, I can do that.

For your question why GHA didn't error is because I used `actions/checkout` with `fetch-depth: 0`. This fetches all branches in the repo. Maybe in your local you don't have the develop branch?

Reply:
Hmmm. I had `develop` in my local repository, but it was a bit old.
I updated the `develop`, and ran `composer update`.  Then, the error is gone.